public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -212,7 +212,9 @@ function getLabel(expr: string, node: SyntaxNode): QueryBuilderLabelFilter {
   const labelNode = node.getChild(Identifier);
   const label = getString(expr, labelNode);
   const op = getString(expr, labelNode!.nextSibling);
-  const value = getString(expr, node.getChild(String)).replace(/"/g, '');
+  let value = getString(expr, node.getChild(String));
+  // `value` is wrapped in double quotes, so we need to remove them. As a value can contain double quotes, we can't use RegEx here.
+  value = value.substring(1, value.length - 1);
```

Comment:
@matyax @ivanahuckova that was the issue which caused wrong visuals in the query builder when switching from code. Since it also messed up my testing I fixed it here directly.